chronos/apps/http/httpclient.nim

```diff
@@ -620,13 +626,23 @@ proc acquireConnection(session: HttpSessionRef,
           raiseHttpConnectionError("Connection timed out")
       res[].state = HttpClientConnectionState.Acquired
       session.connections.mgetOrPut(ha.id, default).add(res)
+      inc(session.connectionsCount)
       return res
 
 proc removeConnection(session: HttpSessionRef,
                       conn: HttpClientConnectionRef) {.async.} =
-  session.connections.withValue(conn.remoteHostname, connections):
-    connections[].keepItIf(it != conn)
+  let removeHost =
+    block:
+      var res = false
+      session.connections.withValue(conn.remoteHostname, connections):
+        connections[].keepItIf(it != conn)
+        if len(connections[]) == 0:
+          res = true
+      res
+  if removeHost:
+    session.connections.del(conn.remoteHostname)
   await conn.closeWait()
+  dec(session.connectionsCount)
```

Comment:
Seems a bit tricky re cancellation of the await above; not sure if there are situations where cancellation can be triggered though.

`connectionsCount` seems to be only used in test code which is a controlled environment, so it's probably alright.